File: backend/src/models/notification.js
const mongoose = require('mongoose')

const notificationSchema = new mongoose.Schema({
    text: {
        type: String
    },
    userName: {
        type: String,
        required: true
    },
    senderId: {
        type: mongoose.Schema.ObjectId,
        ref: 'User',
        required: true
    },
    receiverId: {
        type: mongoose.Schema.ObjectId,
        ref: 'User',
        required: true
    },
    notificationType: {
        type: String,
    }
})

const Notification = mongoose.model("Notification", notificationSchema)

module.exports = Notification

File: frontend/Twitter Clone/Features/Notification/Domain/Notification.swift
//
//  Notification.swift
//  Twitter Clone
//
//  Created by PSG-MDU-HAMZA on 14/09/2026.
//

import Foundation

struct Notification: Decodable {
    var _id: String
    var id: String { _id }
    var text: String?
    var userName: String
    var senderId: String
    var receiverId: String
    var notificationType: NotificationType
}

struct NotificationRequest: Encodable {
    var text: String?
    var userName: String
    var senderId: String
    var receiverId: String
    var notificationType: NotificationType
}

enum NotificationType: String, Codable {
    case like
    case follow
    
    var message: String {
        switch self {
        case .like:
            "Liked your tweet."
        case .follow:
            "Followed you."
        }
    }
}
